### ISSUE TITLE
Not Authorized when signing out

### ISSUE THREAD

**Issue by vigsgb** [IdentityServer/Thinktecture.IdentityServer3#910]
I am having trouble trying to figure out why I am getting an error when I log out of my application calling the endsession with a hint and url redirect.   It appears that my client is not sending along the cookie so the server thinks that I am not logged in.

Strange thing is that everything works fine on my local development machine but when I move it to a development server it no longer works....

So....  I login on my local machine the cookie is passed when I call the end session, no errors and the server redirects me to my log out redirect url.

On development server.... I can login run my app but when I log out the cookie does not get passed and the server throws an error message that I am unauthorized, then cleans up cookies but does not redirect.  I figure the server is designed to not redirect seeing as how it can not verify who was logged in.

Any ideas what might be happening?

Thanks Much,
Greg

**Comment by brockallen** [comment]
Are you sending the user to the correct endpoint? There's the end session endpoint, which is the right one and the endpoint that accepts the id_token_hint. Then it will redirect the user to the logout endpoing (under the authentication controller).

**Comment by vigsgb** [comment]
I am sending the user to:  /connect/endsession?id_token_hint={putting in the id token}&post_logout_redirect_uri={putting in redirect uri}

On my personal development machine everything works fine.  When I run this on another development server with the same code base I see the following error:  w3wp.exe Error: 0 : [Thinktecture.IdentityServer.Core.Validation.EndSessionRequestValidator]: 2/11/2015 10:32:16 PM +00:00 -- User is anonymous. Aborting logout.   

When I look in fiddler I notice that the development environment that is working is sending a cookie along with the endsession request.   The development environment that is not working does not send this cookie on the end session request.   I am trying to figure out why the one server is not sending the cookie.

**Comment by brockallen** [comment]
Perhaps an incorrect host name, so the browser doesn't know it's the same server, and thus isn't sending the cookie.

**Comment by brockallen** [comment]
Any update?

**Comment by vigsgb** [comment]
You were right!

After looking for several hours with Fiddler, IE tools and Chrome tools I was unable to determine why the client was not sending a cookie.   I decided to look once again at the raw log files and found the following:

Request to the identity server was:  https://abc.com/identity/core.....
Response back to the client was:  https://abc.com/Identity/core....

So the reason the cookie was not being sent was that the request had a lower case i in and the response back had an upper case I.   

When looking at the request and responses in Fiddler, Chrome and IE tools all request and responses showed a capital I for all calls.   Only place I noticed the lower case I was in the raw web logs.   

I will add this to my collection of..... Burnt Fingers....  

Thanks for the quick replies and responses....
